nssp_demo/forecast_state.py

```diff
@@ -45,38 +45,68 @@ def postprocess_forecast(model_run_dir: Path) -> None:
     return None
 
 
+def get_available_reports(
+    data_dir: str | Path, glob_pattern: str = "*.parquet"
+):
+    return [
+        datetime.strptime(f.stem, "%Y-%m-%d").date()
+        for f in Path(data_dir).glob(glob_pattern)
+    ]
+
+
 def main(
     disease: str,
     report_date: str,
     state: str,
-    nssp_data_dir: Path | str,
+    facility_level_nssp_data_dir: Path | str,
+    state_level_nssp_data_dir: Path | str,
     param_data_dir: Path | str,
     output_data_dir: Path | str,
     n_training_days: int,
     n_forecast_days: int,
     n_chains: int,
     n_warmup: int,
     n_samples: int,
-    last_training_date: str,
+    exclude_last_n_days: int = 0,
 ):
     logging.basicConfig(level=logging.INFO)
     logger = logging.getLogger(__name__)
 
+    available_facility_level_reports = get_available_reports(
+        facility_level_nssp_data_dir
+    )
+
+    available_state_level_reports = get_available_reports(
+        state_level_nssp_data_dir
+    )
+    first_available_state_report = min(available_state_level_reports)
+    last_available_state_report = max(available_state_level_reports)
+
     if report_date == "latest":
-        report_date = max(
-            f.stem for f in Path(nssp_data_dir).glob("*.parquet")
+        report_date = max(available_facility_level_reports)
+    else:
+        report_date = datetime.strptime(report_date, "%Y-%m-%d").date()
+
+    if report_date in available_state_level_reports:
+        state_report_date = report_date
+    elif report_date > last_available_state_report:
+        state_report_date = last_available_state_report
+    elif report_date > first_available_state_report:
+        raise ValueError(
+            "Dataset appear to be missing some state-level "
+            f"reports. First entry is {first_available_state_report}, "
+            f"last is {last_available_state_report}, but no entry "
+            f"for {report_date}"
         )
-    report_date = datetime.strptime(report_date, "%Y-%m-%d").date()
+    else:
+        state_report_date = None
```

Comment:
This is only reached when `report_date < first_available_state_report`, so why not error instead of None?

Reply:
because in the future we might have facility date that's more out of date than state data (admittedly elsewhere in the code we tacitly assume the facility-level data is more recent.

Reply:
I think I would rather explicitly assume that the facility-level data is more recent so that things aren't silently wrong if that changes.

Reply:
Works for me.

Reply:
Done in e9abdd6